clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -232,20 +232,37 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
       ForeachExt = "aocr";
     }
 
+  StringRef createdReportName;
   for (auto *A : Args) {
-    // Any input file is assumed to have a dependency file associated
-    if (A->getOption().getKind() == Option::InputClass) {
-      SmallString<128> FN(A->getSpelling());
-      StringRef Ext(llvm::sys::path::extension(FN));
-      if (!Ext.empty()) {
-        types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
-        if (Ty == types::TY_INVALID)
-          continue;
-        if (types::isSrcFile(Ty)) {
-          llvm::sys::path::replace_extension(FN, "d");
-          FPGADepFiles.push_back(InputInfo(types::TY_Dependencies,
-              Args.MakeArgString(FN), Args.MakeArgString(FN)));
-        }
+    // Any input file is assumed to have a dependency file associated and
+    // the report folder can also be named based on the first input.
+    if (A->getOption().getKind() != Option::InputClass)
+      continue;
+    SmallString<128> ArgName(A->getSpelling());
+    StringRef Ext(llvm::sys::path::extension(ArgName));
+    if (Ext.empty())
+      continue;
+    types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
+    if (Ty == types::TY_INVALID)
+      continue;
+    if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
+      // Dependency files and the project report are created in CWD, so strip
+      // off any directory information if provided with the input file.
+      // TODO - Use temporary files for dependency file creation and
+      // usage with -fintelfpga.
+      ArgName = llvm::sys::path::filename(ArgName);
+      if (types::isSrcFile(Ty)) {
+        SmallString<128> DepName(ArgName);
+        llvm::sys::path::replace_extension(DepName, "d");
+        FPGADepFiles.push_back(InputInfo(types::TY_Dependencies,
+                                         Args.MakeArgString(DepName),
```

Comment:
What does MakeArgString do?
and why it needs to be passed two times?

Reply:
When creating an Input Type, we need to pass in the name twice - once for the Filename, once for the BaseInput.  see: https://clang.llvm.org/doxygen/classclang_1_1driver_1_1InputInfo.html.  MakeArgString basically constructs a string pointer that lasts as long as the `Args`